lib/elixir/src/elixir_def.erl

```diff
@@ -396,6 +400,10 @@ format_error({clauses_with_defaults, {Kind, Name, Arity}}) ->
     "\n"
    "~ts ~ts/~B has multiple clauses and defines defaults in one or more clauses", [Kind, Name, Arity]);
 
+format_error({grouped_clause, {Kind, Name, _OtherArity, OrigLine, OrigFile}}) ->
+   io_lib:format("clauses with the same name should be grouped together, ~ts ~ts was previously defined (~ts:~B)",
```

Comment:
I think we should wrap the `def foo` part in quotes (it reads a bit confusing in the error message in the test below). Maybe we can include the arity too?

> "def foo/2" was previously defined

Thoughts?

Reply:
I think the arity will help in identifying which clause needs to be corrected and I  agree with you on the quotes let me put in some time and I'll push those fixes

Reply:
I think we can improve both warnings.

Let's call the first `ungrouped_arity` and the other one `ungrouped_name`. For the arity, we should say:

> clauses with the same name and arity (number of arguments) should be grouped together, \"~ts ~ts/~B\" was previously defined (~ts:~B)

For `ungrouped_name`, we will say:

> clauses with the same name should be grouped together, \"~ts ~ts/~B\" was previously defined (~ts:~B)

Reply:
@josevalim  woop woop! yours looks way cooler I like this one